doc/_porting.md

```diff
@@ -31,26 +41,30 @@ In order to make it work, you need to compile it with the following flag
 -DPINOCCHIO_ENABLE_COMPATIBILITY_WITH_VERSION_1
 ```
 
-## Deprecated macros
+\subsection PortingCMacros Deprecated macros
 
 The following marcos are not employed anymore
 ```
 WITH_HPP_FCL
 WITH_URDFDOM
 WITH_LUA5
+URDFDOM_TYPEDEF_SHARED_PTR
 ```
 in favor of
 ```
 PINOCCHIO_WITH_HPP_FCL
 PINOCCHIO_WITH_URDFDOM
 PINOCCHIO_WITH_LUA5
+PINOCCHIO_URDFDOM_TYPEDEF_SHARED_PTR
```

Comment:
This is normally transparent to the users. So do not name it here.

Reply:
I disagree. Users might need to pass this macro as a compilation flag for their code to compile. That's how I discovered it, while porting the `overview-urdf` example in the `doc-v2` branch (see the compilation command). The very fact I discovered it proves it is *not* 100% user-transparent. In fact, it might be a good idea to list here all other macros that were changed, it will not do harm, if you could provide a full list I would be grateful

Reply:
This information is contained inside the pkg file.
This is where you have to find the information for linking Pinocchio.

Reply:
And this is how it has been working for a while with Pinocchio!

Reply:
For instance, `PINOCCHIO_URDFDOM_TYPEDEF_SHARED_PTR`depends on your version of urdfdom package.

Reply:
Then you night want to check the URDF example in topic/doc-v2, because in that example PINOCCHIO_URDFDOM_TYPEDEF_SHARED_PTR is passed explicitly without comments.
Still, I do not see why not documenting the change. These are porting notes meant to ease the migration, the more complete, the better. I repeat, I had to discover it myself in order for that example to work.

Reply:
If the configuration files are meant to be the privileged (if not the only) way of compiling Pinocchio, then they should be properly documented and used throughout the doc, which is not the case now.
Otherwise, if we want to leave open the possibility of compiling without them, and if we are explicitly providing compilation commands, I think these "internal" macros should be documented, including in the porting notes.

Reply:
Managing all the dependencies by hand is a nightmare. 
That's why Cmake or pkg-config are making our lives easier.
It will become a nightmare again to need to document all the examples with precise compilation options which depend on the system used and also the dependencies.
So, I feel against providing this information in the documentation or porting notes.

Reply:
My opinion:
* in the porting notes, we would like to keep track of this changes. They are for expert users, so don't be afraid of it.
* in the doc: the minimal examples are provided with their explicit compilation instruction, following the example of Eigen doc. The remark of @jcarpent justifies the addition in the doc of compilation instruction, in particular explaining how Pinocchio could be easily integrated in a third-party project.

Reply:
But what I meant is that there is no a unique compilation line that will work for sure on all the systems.
So, we should add documentation of them here, we are wasting time and we will never be able to maintain it!

Reply:
If you want to do it correctly, you should use `pkg-config` command line to retrieve the dependencies and the compilation flags. This is the only way to be generic!!!

Reply:
By the way, @nmansard and @gabrielebndn you should check this package here: https://github.com/stack-of-tasks/pinocchio-minimal
That has been done to show how to use Pinocchio in an external project!!!

Reply:
I knew that package, thanks. But in the doc the example just gave the compilation command and that's how I stumbled on the problem.

Reply:
But they are not generic. This means that they will work for your configuration but not on my computer for instance.